core/state-db/src/noncanonical.rs

```diff
@@ -112,9 +116,35 @@ impl<BlockHash: Hash, Key: Hash> NonCanonicalOverlay<BlockHash, Key> {
 			parents,
 			pending_canonicalizations: Default::default(),
 			pending_insertions: Default::default(),
+			values: values,
 		})
 	}
 
+	fn insert_values(values: &mut HashMap<Key, (u32, DBValue)>, inserted: Vec<(Key, DBValue)>) {
```

Comment:
free functions are more idiomatic

Reply:
Not sure why. Member function have a benefit of using generic types that the struct is parameterized with, without having to specify them again along with the trait bounds, etc.